core/src/main/java/org/apache/accumulo/core/client/ScannerBase.java

```diff
@@ -370,4 +372,15 @@ default void forEach(BiConsumer<? super Key,? super Value> keyValueConsumer) {
       keyValueConsumer.accept(entry.getKey(), entry.getValue());
     }
   }
+
+  /**
+   * Stream the Scanner results sequentially from this scanner's iterator
+   *
```

Comment:
Should we add a note here that if the resulting stream cannot be used outside of a try-with-resources block that closes the Scanner and that if the stream is returned from a function without the Scanner being closed then `.onClose(scanner::close)` should be called before leaving the method?

Reply:
I don't think that's necessary. We shouldn't specifically require the use of a try-with-resources block to close the Scanner. That's an optional language construct. The same goes for using `.onClose(scanner::close)`. It's a good idea for the user to close the Scanner object when they are done with it, but we shouldn't be prescriptive about how they manage their resource. They can close it whenever they like. They may want to re-use the Scanner for another stream.

We don't offer recommendations for closing Scanner when iterator() or forEach() is used. I don't see why this would be different. It should suffice that Scanner is Closeable, and therefore expected to be closed when they are finished using it.

Reply:
Will merge as-is, but feel free to follow up with additional javadoc changes or other docs if you feel it would be helpful.